src/studio/src/designer/frontend/ux-editor/containers/DroppableDraggableComponent.tsx

```diff
@@ -1,316 +1,101 @@
-/* eslint-disable react/no-find-dom-node */
-import React from 'react';
+import React, { memo, RefObject, useRef, useState } from 'react';
 import {
-  ConnectDragPreview,
-  ConnectDragSource,
-  ConnectDropTarget,
-  DragSource,
-  DragSourceConnector,
-  DragSourceMonitor,
-  DragSourceSpec,
-  DropTarget,
-  DropTargetConnector,
+  DropTargetHookSpec,
   DropTargetMonitor,
-  DropTargetSpec,
+  useDrag,
+  useDrop,
 } from 'react-dnd';
-import ReactDOM from 'react-dom';
-
-export interface IDroppableDraggableComponentProps {
-  id: string;
-  index: number;
-  containerId: string;
-  canDrag: boolean;
-  onMoveComponent?: (...args: any) => void;
-  onDropComponent?: (...args: any) => void;
-  onMoveContainer?: (...args: any) => void;
-  onDropContainer?: (...args: any) => void;
-}
-
-const dragSourceSpec: DragSourceSpec<IDroppableDraggableComponentProps, any> = {
-  beginDrag(props: any) {
-    return {
-      ...props,
-    };
-  },
-
-  isDragging(
-    props: IDroppableDraggableComponentProps,
-    monitor: DragSourceMonitor,
-  ) {
-    return props.id === monitor.getItem().id;
-  },
-  canDrag(props: IDroppableDraggableComponentProps) {
-    return props.canDrag;
-  },
-  endDrag(
-    props: IDroppableDraggableComponentProps,
-    monitor: DragSourceMonitor,
-    component: any,
-  ) {
-    if (!monitor.didDrop() && component) {
-      const draggedComponent = monitor.getItem();
-      props.onDropComponent(
-        draggedComponent.id,
-        draggedComponent.containerId,
-        props.containerId,
-      );
+import { dragSourceSpec, hoverIndexHelper } from './helpers/dnd-helpers';
+import { EditorDndEvents, EditorDndItem, ItemType } from './helpers/dnd-types';
+import classNames from 'classnames';
+
+const dropTargetSpec = (
```

Comment:
A bit difficult to tell because of how the diff is rendered, but is this the same code as in `src/studio/src/designer/frontend/ux-editor/containers/DroppableDraggableContainer.tsx`? From what I can tell it seems to be similar at least.

There are some redundant comments that can be removed here as well

Reply:
Det er den samme komponenten. Har prøvd å beholde strukturen, slik at det skal være greit å kjenne seg igjen. Vet ikke hvor godt det lyktes.